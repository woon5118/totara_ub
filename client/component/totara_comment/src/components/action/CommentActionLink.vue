<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Kian Nguyen <kian.nguyen@totaralearning.com>
  @module totara_comment
-->
<template>
  <div class="tui-commentActionLink">
    <a
      v-if="linkText"
      href="#"
      class="tui-commentActionLink__link"
      :class="{
        'tui-commentActionLink__link--small': isSmall,
        'tui-commentActionLink__link--disabled': loading || !showLoadMore,
      }"
      @click.prevent="handleClick"
    >
      <span>{{ linkText }}</span>
    </a>
  </div>
</template>

<script>
import { isValid, SIZE_SMALL } from 'totara_comment/index';

export default {
  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    showLoadMore: {
      type: Boolean,
      default: false,
    },

    size: {
      type: String,
      default() {
        return SIZE_SMALL;
      },

      validator(prop) {
        return isValid(prop);
      },
    },
  },

  computed: {
    isSmall() {
      return SIZE_SMALL === this.size;
    },

    linkText() {
      if (this.showLoadMore) {
        return this.$str('loadmorecomments', 'totara_comment');
      }

      return '';
    },
  },

  methods: {
    handleClick() {
      if (this.loading) {
        return;
      }

      if (this.showLoadMore) {
        this.$emit('load-more');
      }
    },
  },
};
</script>

<lang-strings>
  {
    "totara_comment": [
      "loadmorecomments"
    ]
  }
</lang-strings>

<style lang="scss">
.tui-commentActionLink {
  &__link {
    &--small {
      @include tui-font-link-small();
    }

    &--disabled {
      color: var(--color-neutral-5);
      cursor: not-allowed;

      &:hover,
      &:focus {
        color: var(--color-neutral-5);
        text-decoration: none;
      }
    }
  }
}
</style>
