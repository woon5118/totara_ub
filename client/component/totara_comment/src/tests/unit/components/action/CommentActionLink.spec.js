/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Kian Nguyen <kian.nguyen@totaralearning.com>
 * @module totara_comment
 */
import CommentActionLink from 'totara_comment/components/action/CommentActionLink';
import { shallowMount } from '@vue/test-utils';

describe('totara_comment/components/action/CommentActionLink.vue', () => {
  let wrapper = null;

  beforeAll(() => {
    wrapper = shallowMount(CommentActionLink, {
      propsData: {
        text: 'Some random txt',
      },

      mocks: {
        $str(x, y) {
          return `${x}-${y}`;
        },
      },
    });
  });

  it('Checks snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
