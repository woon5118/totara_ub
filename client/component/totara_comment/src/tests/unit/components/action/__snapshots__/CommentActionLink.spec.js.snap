// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`totara_comment/components/action/CommentActionLink.vue Checks snapshot 1`] = `
<div
  class="tui-commentActionLink"
  text="Some random txt"
>
  <!---->
</div>
`;
